custom/GLideN64/mupenplus/CommonAPIImpl_mupenplus.cpp

```diff
@@ -94,9 +99,9 @@ void PluginAPI::FindPluginPath(wchar_t * _strPath)
 		_getWSPath(path, _strPath);
 	}
 #elif defined(OS_MAC_OS_X)
-	char path[MAXPATHLEN];
-	uint32_t pathLen = MAXPATHLEN * 2;
-	if (_NSGetExecutablePath(path, pathLen) == 0) {
+	char path[PATH_MAX_LENGTH];
+	uint32_t pathLen = PATH_MAX_LENGTH * 2;
```

Comment:
This one is bothersome. Is this a out pointer?
If its a max len then it could write double of path_max_length to the string however by design _NSGetExecutablePath should never write more than that anyway i assume?

Reply:
A pointer to pathLen is passed yes. I don't know why the code used to multiply this by 2.
I can remove it and see what happen.

Reply:
Is it possible that 2 is the size of char here?

Reply:
Yeah this is a buffer size that we pass as a pointer, so to handle unicode we might need to *2, that's the only explanation I can find.